### ISSUE TITLE
Ensure `Value_Type.Byte` behaves consistently between in-memory and Database

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#7888]
We have a `Value_Type.Byte` which is a special type used to represent small 8-bit integers.

Most database systems will treat such type as an `unsigned byte` with the range of values being `0-255`. However, our in-memory backend currently relies on Java's defaults where all numeric types are signed and the `byte` values are actually `-128` to `127`. This is not consistent with databases.

-  [ ] We should modify our implementation to ensure that `Value_Type.Byte` actually represents the `0-255` range.
	- With the current non-compact storage that only means amending the bounds. In the future, if we will want a more compact storage, we may need to ensure a proper mapping when storing/retrieving the values (so that we can store them as 8-bit `byte` type but that Enso will see it as `Integer` in range `0-255`).
- [ ] We should ensure that when fetching a Byte column from a Database or uploading a Byte column to Database all values are mapped correctly.
- [ ] We should ensure that the documentation for `Value_Type.Byte` defines how it is supposed to behave (e.g. `0-255` regardless of backend/platform).

**Comment by radeusgd** [comment]
I think this task is kind of blocked by waiting for new database integrations.

That is because our current two backends do not really support a `Byte` type:
- SQLite just stores integers without limits, so we map everything to `Integer 64-bit`.
- Postgres does [not have a numeric byte type](https://dba.stackexchange.com/a/159916), only a [byte array](https://www.postgresql.org/docs/current/datatype-binary.html).

So I think it will only matter once we actually have a Database that can store 8-bit integers, e.g. [SQL Server that has `tinyint`](https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16).

-----

It may make sense to tackle the task earlier to ensure we don't introduce a breaking change later (changing the range on `Value_Type.Byte` from current `-128` to `127` to `0-255`), though.

**Comment by radeusgd** [comment]
Just changing the 'working range' should actually be trivial with the current implementation that stores everything as a `long`. It can easily be done in probably less than half a day. So we may consider doing this change right away and then doing a separate integration once we have a `Byte` supporting DB backend.

**Comment by jdunkerley** [comment]
Need to review that round trip is possible.

Should get the value from a byte column as a 0-255 (stored as a long behind the scenes).
Need to be able to write back to same DB tech with same type.

**Comment by jdunkerley** [comment]
It is on hold as no backend currently supported which has a byte type.
